web/src/feedback_widget.ts

```diff
@@ -22,15 +22,33 @@ Code-wise it's a singleton widget that controls the DOM inside
 
 */
 
-const meta = {
-    hide_me_time: null,
+type Meta = {
+    undo?: () => void;
+    hide_me_time: number;
+    alert_hover_state: boolean;
+    $container?: JQuery | undefined;
+    opened: boolean;
+    handlers_set?: boolean;
+};
+
+type Populate = {
+    title_text: string;
+    undo_button_text: string;
+    populate: (container: JQuery) => void;
+    on_undo: () => void;
+};
```

Comment:
Like I said above:

> Also, both of these types will only ever be referenced from a single declaration. There isn’t any reason to name them from anywhere other than that declaration, so they should just be inlined into the declaration.

We should just write `const meta: { hide_me_time: number, … } = { hide_me_time: 0, … }` and `function show(opts: { title_text: string, … }): void`.